fix: search projects title

disha1202 opened the pull request:
Fixes #4122 

#### Describe the changes you have made in this PR -

1. Use ellipse overflow
2. Show full name in tooltip

### Screenshots of the changes (If any) -

<img width="1440" alt="image" src="https://github.com/CircuitVerse/CircuitVerse/assets/58461801/b811b552-c4f4-4baa-86a9-8e92e5839ece">


Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

disha1202 commented:
@vedant-jain03 @tanmoysrt 
Please take a look. 
Also I haven't added screenshots as I am unable to test it on local.

vedant-jain03 commented:
> @vedant-jain03 @tanmoysrt Please take a look. Also I haven't added screenshots as I am unable to test it on local.

Thanks so how have you tested the changes @disha1202 ?

tanmoysrt commented:
> > > @vedant-jain03 @tanmoysrt Please take a look. Also I haven't added screenshots as I am unable to test it on local.
> > 
> > 
> > Thanks so how have you tested the changes @disha1202 ?
> 
> I haven't tested it on local though I took reference from the [existing code](https://github.com/CircuitVerse/CircuitVerse/blob/e73103de5041d5f94fe38833233c59deeab6e198/app/assets/stylesheets/users.scss#L286) which working perfectly fine.
> 
> Hoping this would work as well :)
> 
> PS: If there is any way to test it locally please let me know.
Checkout the [Setup Documentation](https://github.com/CircuitVerse/CircuitVerse/blob/master/SETUP.md)
Try out `Github Codespaces` or `GitPod` to setup and run quickly.

disha1202 commented:
> > > > @vedant-jain03 @tanmoysrt Please take a look. Also I haven't added screenshots as I am unable to test it on local.
> > > 
> > > 
> > > Thanks so how have you tested the changes @disha1202 ?
> > 
> > 
> > I haven't tested it on local though I took reference from the [existing code](https://github.com/CircuitVerse/CircuitVerse/blob/e73103de5041d5f94fe38833233c59deeab6e198/app/assets/stylesheets/users.scss#L286) which working perfectly fine.
> > Hoping this would work as well :)
> > PS: If there is any way to test it locally please let me know.
> 
> Checkout the [Setup Documentation](https://github.com/CircuitVerse/CircuitVerse/blob/master/SETUP.md#installation) Try out `Github Codespaces` or `GitPod` to setup and run quickly.

Thanks @tanmoysrt
Updated the code and also added screenshot.

vedant-jain03 commented on app/assets/stylesheets/search.scss:
```diff
@@ -76,12 +76,18 @@
 }
 
-.search-project-name {
+.search-project-name h3 {
```
Try this:
```
.search-project-name {
  height: 40px;
  text-align: left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.search-project-name h3 {
  display: inline;
}
```

Why?
- We make overflow to parent of the text, in the case `h3` is block element, so we make it inline to ensure that ellipse works fine for that
- With the initial code, not able to understand why you made it `70%` but with the initial code, even the name is `Public` it is getting ellipsed which ideally should not happen

<img width="1174" alt="Screenshot 2023-10-09 at 9 38 49 PM" src="https://github.com/CircuitVerse/CircuitVerse/assets/76901313/0e7138e6-86a2-4059-9c21-5fff17e19898">

vedant-jain03 reviewed (changes requested):
So for the name "Public", the text is ellipsed which ideally should not happen

<img width="1174" alt="Screenshot 2023-10-09 at 9 38 49 PM" src="https://github.com/CircuitVerse/CircuitVerse/assets/76901313/66efa7b2-848f-4bcf-98b7-37dadc093996">

Possible solution:
- Make parent

disha1202 commented:
Hi @vedant-jain03 
Thanks for reviewing, I have updated the PR as per your suggestions. 
Please take a look.
